src/main/java/com/cloudbees/jenkins/plugins/mtslavescloud/MansionNodeProvisionerStrategy.java

```diff
@@ -0,0 +1,172 @@
+     * Returns {@code true} if this strategy is enabled for the current Jenkins instance.
+     *
+     * @return {@code true} if this strategy is enabled for the current Jenkins instance.
+     */
+    public static boolean isEnabled() {
+        MansionNodeProvisionerStrategy strategy = getInstance();
+        return strategy != null && strategy.enabled;
+    }
+
+    /**
+     * Sets whether this strategy is enabled for the current Jenkins instance. Useful to disable this strategy
+     * in groovy console
+     *
+     * @param enabled if {@code true} then the strategy will be enabled - including injecting it in the list of
+     *                strategies if necessary, if {@code false} then the strategy will be disabled.
+     */
+    public static void setEnabled(boolean enabled) {
+        MansionNodeProvisionerStrategy strategy = getInstance();
+        if (strategy == null && enabled) {
+            strategy = new MansionNodeProvisionerStrategy();
+            ExtensionList.lookup(NodeProvisioner.Strategy.class).add(0, strategy);
+        }
+        if (strategy != null) {
+            strategy.enabled = enabled;
+        }
+    }
+
+    @Nonnull
+    @Override
+    public NodeProvisioner.StrategyDecision apply(@Nonnull NodeProvisioner.StrategyState strategyState) {
+        final Label label = strategyState.getLabel();
+
+        final SlaveTemplate st = SlaveTemplateList.get().get(label);
+        if (st == null) {
+            return NodeProvisioner.StrategyDecision.CONSULT_REMAINING_STRATEGIES;
+        }
+        if (!st.isEnabled()) {
+            LOGGER.log(Level.FINE, "Slave template is disabled {0}", st);
+            return NodeProvisioner.StrategyDecision.CONSULT_REMAINING_STRATEGIES;
+        }
+
+        final MansionCloud mansionCloud = getCloudImpl();
+        if (mansionCloud == null) {
+            return NodeProvisioner.StrategyDecision.CONSULT_REMAINING_STRATEGIES;
+        }
+        if (mansionCloud.getBackOffCounter(st).isBackOffInEffect()) {
+            LOGGER.log(Level.FINE, "Back off in effect for {0}", st);
+            return NodeProvisioner.StrategyDecision.CONSULT_REMAINING_STRATEGIES;
+        }
+        final HardwareSpec box = mansionCloud.getBoxOf(st, label);
+        if (mansionCloud.getQuotaProblems().isBlocked(box, st)) {
+            LOGGER.log(Level.FINE, "Provisioning of {0} blocked by quota problems.", st);
+            return NodeProvisioner.StrategyDecision.CONSULT_REMAINING_STRATEGIES;
+        }
+        LoadStatistics.LoadStatisticsSnapshot snapshot = strategyState.getSnapshot();
+        int availableCapacity = snapshot.getAvailableExecutors() + snapshot.getConnectingExecutors() +
+                strategyState.getAdditionalPlannedCapacity();
+        int currentDemand = snapshot.getQueueLength();
+        LOGGER.log(Level.FINE, "Available capacity={0}, currentDemand={1}",
+                new Object[]{availableCapacity, currentDemand});
+        if (availableCapacity < currentDemand) {
```

Comment:
> To be clear, what we are doing here is saying that provisioning a new node from mansion-cloud should always be preferred over re-using an existing provisioned node when it becomes free.

Yep, that's what we want, one shot slave.

> I suspect that we actually want something that takes the current number of provisioned nodes into account, e.g. 

There'a a guard in  https://github.com/jenkinsci/mansion-cloud-plugin/pull/9/files#diff-65edda8ebfeb0e7569ec6b9cf2328232R272 to prevent creating too much slaves at the same time.
My idea was to call `MansionCloud.provision()` every time there's some slaves to create and let it decide if it really creates the nodes.